TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -214,15 +214,37 @@ public override bool Visit(Move move)
                 {
                     return true;
                 }
+
+                var senderIsAlphanumeric = false;
+                DataDefinition senderDataDefinition = null;
+                if (moveSimple.SendingVariable?.StorageArea?.Kind == StorageAreaKind.DataOrCondition
+                    && move.StorageAreaReadsDataDefinition?.TryGetValue(moveSimple.SendingVariable.StorageArea, out senderDataDefinition) == true)
+                {
+                    senderIsAlphanumeric = senderDataDefinition.DataType == DataType.Alphanumeric;
+                }
+
                 foreach (var area in moveSimple.StorageAreaWrites)
                 {
                     var receiver = area.StorageArea;
                     if (receiver is FunctionCallResult)
                         DiagnosticUtils.AddError(move, "MOVE: illegal <function call> after TO");
+
+                    if (senderIsAlphanumeric
+                        && receiver is DataOrConditionStorageArea dataOrConditionStorageArea
```

Comment:
Why the cast ? If testing is required, you can check the kind just like the sender.

Reply:
It was done like this above, but the cast does not seem to be required in both cases, changing it to test on kind.